packages/image_picker/image_picker_android/android/src/main/java/io/flutter/plugins/imagepicker/ImageResizer.java

```diff
@@ -112,14 +131,43 @@ private void copyExif(String filePathOri, String filePathDest) {
     exifDataCopier.copyExif(filePathOri, filePathDest);
   }
 
-  private Bitmap decodeFile(String path) {
-    return BitmapFactory.decodeFile(path);
+  private SizeFCompat readFileDimensions(String path) {
+    BitmapFactory.Options options = new BitmapFactory.Options();
+    options.inJustDecodeBounds = true;
+    decodeFile(path, options);
+    return new SizeFCompat(options.outWidth, options.outHeight);
+  }
+
+  private Bitmap decodeFile(String path, @Nullable BitmapFactory.Options opts) {
+    return BitmapFactory.decodeFile(path, opts);
   }
 
   private Bitmap createScaledBitmap(Bitmap bmp, int width, int height, boolean filter) {
     return Bitmap.createScaledBitmap(bmp, width, height, filter);
   }
 
+  /**
+   * Calculates the largest sample size value that is a power of two based on a target width and
+   * height.
```

Comment:
Please link to https://developer.android.com/topic/performance/graphics/load-bitmap#load-bitmap to provide context for why that's the logic being used to determine the sample size.